### PR TITLE
Technology Updates - 8th, 22nd March 2022

### PR DESCRIPTION
**Pull request by siforster**
Updates to technology document as a result of reviews on 8th, 22nd March

### CODE REVIEW

**Review by daniellemcook — DISMISSED** [review]

File: reference/technology.md [hunk 1]
```diff
@@ -10,29 +10,27 @@ The Cloud Native Maturity Model is composed of five separate documents - the [Pr
...
-You’ll probably have had to incorporate monitoring and observability into your workloads. You’ll have brought Prometheus in and started monitoring your clusters for standard metrics such as RAM, CPU etc.  
-While you might be starting to evaluate application tracing, don’t worry about it too much if you have started to gather core metrics.
-Your focus here is on getting an application running in production and having enough to support it.
+* Level 2: This marks your first step into production. You’ve worked hard to build your foundation in Level 1, and now you are moving to production. You might have started with something relatively small and simple, but this leap to production has certainly required you to address some significant steps. You’ll probably have had to incorporate monitoring and observability into your workloads. You’ll have brought Prometheus in and started monitoring your clusters for standard metrics such as RAM, CPU etc. While you might be starting to evaluate application tracing, don’t worry about it too much if you have started to gather core metrics. Your focus here is on getting an application running in production and having enough to support it.
```
[the remark this answers is not in the document]
Comment (siforster) [reply reference/technology.md:33]: Amended to include "platform resource, observability and operational capability to support it within your organization"

File: reference/technology.md [hunk 2]
```diff
@@ -42,13 +40,13 @@ Your focus here is on getting an application running in production and having en
...
+* Level 1: You are building your cloud infrastructure either on-prem or off. It will pay dividends to consider early your supporting technology such as your network, firewalls and IAM, access controls and policies (and if you need to change them). Many topics will come out of your initial experimentation with Kubernetes, so ensure you keep track of these - they are the ‘breadcrumbs’ you will follow as you move towards cloud native. This will include RBAC policies, load balancer and/or ingress configuration, cluster dashboards, privileged access (or lack thereof!) and container logging. Your aim is to move away from ‘pets’ to ‘livestock’ so you invest in declarative solutions for your Infrastructure as a Service with Infrastructure as Code (IaC) tooling. If you do not have a consolidated DevOps practice at this level, bring your future operations team in at this level to build familiarity.
```
[the remark this answers is not in the document]
Comment (siforster) [reply reference/technology.md:43]: deleted final instance of "at this level"

File: reference/technology.md [hunk 3]
```diff
@@ -42,13 +40,13 @@ Your focus here is on getting an application running in production and having en
...
 
-* Level 2: Because production is your goal, you’ve built Kubernetes clusters for production. Your aim is to move away from ‘pets’ to ‘livestock’ so you invest in declarative solutions for your Infrastructure as a Service with Infrastructure as Code (IaC) tooling such as Terraform, Pulumi etc.
+* Level 2: Because production is your goal, you’ve built Kubernetes clusters for production.
```
[the remark this answers is not in the document]
Comment (siforster) [reply reference/technology.md:45]: Added.

File: reference/technology.md [hunk 4]
```diff
@@ -42,13 +40,13 @@ Your focus here is on getting an application running in production and having en
...
+* Level 3: As part of building confidence in your cloud infrastructure, you need to gain visibility into what your infrastructure is doing. Developing your monitoring, alerting and resource usage capabilities is going to be your focus. An important consideration here is that where previously you may have considered machine-specific properties such as cpu, ram etc, you’ll also want to factor in cluster resource metrics also. Additionally, you’ll replace components when they fail rather than spending time remediating issues in production. This builds on Level two. Furthermore, you may also look to manage infrastructure with Kubernetes like you would software.
```
[the remark this answers is not in the document]
Comment (siforster) [reply reference/technology.md:47]: Upper cased CPU and RAM, also changed 'two' to '2'

File: reference/technology.md [hunk 5]
```diff
@@ -58,60 +56,62 @@ Start with a canonical microservice application if you can and confirm that it r
...
 
-* Level 3: Whereas in level 2 you’ve been experimenting, in level 3, as you increase your workload, and as you scale, you need consistent tooling across clusters to gain continuous visibility into your Kubernetes clusters. This should include automatic scanning and having runtime observability of what is occurring within your containers and your cluster. CNCF projects such as Falco, Kyverno, and OPA are good options here. You will have alerting and dashboards in place.
+* Level 3: Whereas in level 2 you’ve been experimenting, in level 3, as you increase your workload, and as you scale, you need consistent tooling across clusters to gain continuous visibility into your Kubernetes clusters. This should include automatic scanning and having runtime observability of what is occurring within your containers and your cluster. CNCF projects are good options here. You will have alerting and dashboards in place.
```
[the remark this answers is not in the document]
Comment (siforster) [reply reference/technology.md:75]: fixed.

File: reference/technology.md [hunk 6]
```diff
@@ -58,60 +56,62 @@ Start with a canonical microservice application if you can and confirm that it r
...
 
-* Level 4: With your sources of information you've gained from level three, your goal is to further integrate your data sources and gain visibility along with alerting. This closes the feedback loop on runtime and operations and allows you to respond quickly to abnormal events.
+* Level 4: With your sources of information you've gained from level three, your goal is to further integrate your data sources and gain visibility along with alerting. This closes the feedback loop on runtime and operations and allows you to respond quickly to unplanned events.
```
[the remark this answers is not in the document]
Comment (siforster) [reply reference/technology.md:77]: fixed

File: reference/technology.md [hunk 7]
```diff
@@ -58,60 +56,62 @@ Start with a canonical microservice application if you can and confirm that it r
...
 
 * Level 5: You’re now automating the response to events, and you have all your security data in one central repository. The platform is able to respond to events
 
 ## Application Release and Operations
 
+Managing a cluster with infrastructure-as-code is different to managing application release and deployment, however many of the same techniques and tools will be common to both.
```
[the remark this answers is not in the document]
Comment (siforster) [reply reference/technology.md:83]: Agreed.  Updated to 'Infrastructure as Code'.  Also appended (IaC) as this the allows for further usage later on having introduced it.  Further, allows for consistency with Glossary.

File: reference/technology.md [hunk 8]
```diff
@@ -58,60 +56,62 @@ Start with a canonical microservice application if you can and confirm that it r
...
+* Level 2: Ensure that your development and operations groups are following good practice with containers, secrets and security. Because you are in production, you will want to ensure that you have encryption as well and authentication and authorization addressed.
 
-* Level 3: It’s now time to step up and automate your deployment guardrails and security best practices with policy as code Kyverno and Open Policy Agent are good options here.
+* Level 3: It’s now time to step up and automate your deployment guardrails and security best practices with policy as code. Determine your strategy for enforcement. Begin adopting third party benchmarks and standards where relevant. Consider also adopting anomaly and threat detection technologies.
 
-* Level 4: Here you can attempt  dry-runs of your policy against production in case you haven’t don’t this already. You’ll continue to tune your policies in production.
+* Level 4: Apply your policy against production in case you haven’t already. You’ll continue to tune your policies in production.
```
[the remark this answers is not in the document]
Comment (siforster) [reply reference/technology.md:115]: I have logged issue 17 for this:
https://github.com/cncf/cartografos/issues/17
Great suggestion and suitable topic for community meeting.

File: reference/technology.md [hunk 9]
```diff
@@ -10,29 +10,27 @@ The Cloud Native Maturity Model is composed of five separate documents - the [Pr
 * [Technology](./technology.md)
 
 ## Position on Included Technologies
-***The Cloud Native Maturity model includes references to only CNCF graduated or incubating projects. The Maturity Model’s default position on CNCF sandbox projects will be to exclude unless referenced in later stages of maturity (i.e. users that have achieved level 4 or 5). It does not and will not include any reference to commercial software.*** 
+
+***The Cloud Native Maturity model includes references to only CNCF graduated or incubating projects. The Maturity Model’s default position on CNCF sandbox projects will be to exclude unless referenced in later stages of maturity (i.e. users that have achieved level 4 or 5). It does not and will not include any reference to commercial software.***
```
[the remark this answers is not in the document]
Comment (siforster) [reply reference/technology.md:14]: Fixed

File: reference/technology.md [hunk 10]
```diff
@@ -58,60 +56,62 @@ Start with a canonical microservice application if you can and confirm that it r
...
+* Level 4: As your environment becomes more complex in production, some issue remediation may require adjusting your policy-as-code or components of your infrastructure-as-code, as well as your application. Issues may relate to more than one application so you will aggregate across applications to determine trends. These may relate to bugs such as memory leaks, as well as security or policy issues. Your remediation may be to fix them at source, ideally as ‘far left’ as possible, or otherwise building automation capable of fixing them when they occur, and tuning it over time.
```
Comment (Pinolo) [inline reference/technology.md:103]: there's an "infrastructure-as-code" leftover in the first period. s/infrastructure-as-code/Infrastructure as Code/

File: reference/technology.md [hunk 11]
```diff
@@ -58,60 +56,62 @@ Start with a canonical microservice application if you can and confirm that it r
...
 
-* Level 3: Whereas in level 2 you’ve been experimenting, in level 3, as you increase your workload, and as you scale, you need consistent tooling across clusters to gain continuous visibility into your Kubernetes clusters. This should include automatic scanning and having runtime observability of what is occurring within your containers and your cluster. CNCF projects such as Falco, Kyverno, and OPA are good options here. You will have alerting and dashboards in place.
+* Level 3: Whereas in Level 2 you’ve been experimenting, in level 3, as you increase your workload, and as you scale, you need consistent tooling across clusters to gain continuous visibility into your Kubernetes clusters. This should include automatic scanning and having runtime observability of what is occurring within your containers and your cluster. CNCF projects are good options here. You will have alerting and dashboards in place.
```
Comment (Pinolo) [inline reference/technology.md:75]: s/level 3/Level 3/

**Review by Pinolo — CHANGES_REQUESTED** [review]

File: reference/technology.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (Pinolo) [reply reference/technology.md:115]: Yes, let's talk about this in the community meeting

File: reference/technology.md (hunk 10 above)
Comment (siforster) [reply reference/technology.md:103]: fixed!

**Pull request merged**